File: CiaiControllerSDK/Logging/LoggerProvider.cs
using Microsoft.Extensions.Logging;

namespace CiaiControllerSDK.Logging
{
    /// <summary>
    /// 日志管理器 - 提供全局日志工厂和静态日志访问
    /// </summary>
    public static class LoggerProvider
    {
        private static ILoggerFactory _factory;
        private static readonly object _lock = new();

        /// <summary>
        /// 获取当前日志工厂
        /// </summary>
        public static ILoggerFactory Factory
        {
            get
            {
                if (_factory == null)
                {
                    lock (_lock)
                    {
                        if (_factory == null)
                        {
                            // 默认使用 ConsoleLogger
                            _factory = LoggerFactory.Create(builder =>
                            {
                                builder.AddConsole();
                                builder.SetMinimumLevel(LogLevel.Debug);
                            });
                        }
                    }
                }
                return _factory;
            }
        }

        /// <summary>
        /// 设置自定义日志工厂
        /// </summary>
        /// <param name="factory">日志工厂实例</param>
        /// <remarks>
        /// 调用方可通过此方法注入自定义的日志实现（如 Serilog、NLog 等）
        /// 注意：调用方需自行管理工厂的生命周期，不要在使用期间 dispose
        /// 示例：
        /// <code>
        /// var loggerFactory = LoggerFactory.Create(builder =>
        ///     builder.AddSerilog(new LoggerConfiguration()
        ///         .WriteTo.File("log.txt")
        ///         .CreateLogger()));
        /// LoggerProvider.SetLoggerFactory(loggerFactory);
        /// </code>
        /// </remarks>
        public static void SetLoggerFactory(ILoggerFactory factory)
        {
            lock (_lock)
            {
                // 不释放旧的工厂，由调用方管理生命周期
                _factory = factory;
            }
        }

        /// <summary>
        /// 创建类型化日志器
        /// </summary>
        /// <typeparam name="T">类型</typeparam>
        /// <returns>日志器实例</returns>
        public static ILogger CreateLogger<T>()
        {
            return Factory.CreateLogger<T>();
        }

        /// <summary>
        /// 创建命名日志器
        /// </summary>
        /// <param name="categoryName">类别名称</param>
        /// <returns>日志器实例</returns>
        public static ILogger CreateLogger(string categoryName)
        {
            return Factory.CreateLogger(categoryName);
        }

        /// <summary>
        /// 重置为默认的 ConsoleLogger
        /// </summary>
        public static void ResetToDefault()
        {
            lock (_lock)
            {
                // 不释放旧的工厂，由调用方管理生命周期
                _factory = LoggerFactory.Create(builder =>
                {
                    builder.AddConsole();
                    builder.SetMinimumLevel(LogLevel.Debug);
                });
            }
        }
    }
}

File: CiaiControllerSDKForJava/src/main/java/com/ciai/controller/sdk/logging/LoggerProvider.java
package com.ciai.controller.sdk.logging;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * 日志提供器
 */
public final class LoggerProvider {

    private static org.slf4j.ILoggerFactory loggerFactory;

    static {
        // 默认使用SLF4J的默认实现
        loggerFactory = LoggerFactory.getILoggerFactory();
    }

    private LoggerProvider() {
        // 私有构造函数，防止实例化
    }

    /**
     * 获取日志工厂
     */
    public static org.slf4j.ILoggerFactory getFactory() {
        return loggerFactory;
    }

    /**
     * 设置自定义日志工厂
     */
    public static void setLoggerFactory(org.slf4j.ILoggerFactory factory) {
        if (factory != null) {
            loggerFactory = factory;
        }
    }

    /**
     * 创建日志器
     */
    public static Logger createLogger(Class<?> clazz) {
        return loggerFactory.getLogger(clazz.getName());
    }

    /**
     * 创建日志器
     */
    public static Logger createLogger(String categoryName) {
        return loggerFactory.getLogger(categoryName);
    }

    /**
     * 重置为默认
     */
    public static void resetToDefault() {
        loggerFactory = LoggerFactory.getILoggerFactory();
    }
}
